burn-import/src/onnx/op_configuration.rs

```diff
@@ -453,17 +433,76 @@ pub fn reshape_config(node: &Node) -> Vec<i64> {
         panic!("Zero shape size is not supported");
     }
 
-    let shape = match node.states.first() {
-        Some(state) => match &state.ty {
-            StateType::Tensor(tensor) => match tensor.data.as_ref() {
-                Some(TensorData::Int64(data)) => data.clone(),
-                _ => panic!("Reshape: invalid state data for shape"),
-            },
-        },
-        None => panic!("Reshape: missing state required for shape"),
-    };
+    if node.inputs.len() != 2 || node.inputs[1].value.is_none() {
+        panic!("Reshape: shape tensor must be present");
+    }
+
+    let input_value = &node.inputs[1].value;
+    match &node.inputs[1].ty {
+        ArgType::Tensor(tensor) => {
+            assert_eq!(tensor.dim, 1, "Reshape: shape tensor must be 1D");
+
+            if let Some(Data::Int64s(shape)) = input_value.as_ref() {
+                shape.clone()
+            } else {
+                panic!("Tensor data type must be int64")
+            }
+        }
+        _ => panic!("Only tensor input is valid for shape"),
+    }
+}
+
+pub fn clip_config(node: &Node) -> (Option<f64>, Option<f64>) {
+    let mut min_result: Option<f64> = None;
+    let mut max_result: Option<f64> = None;
+
+    // For Clip Opset 6+ , the min and max values are attributes
+    for (key, value) in node.attrs.iter() {
+        match key.as_str() {
+            "min" => {
+                let min = value.clone().into_f32() as f64;
+                min_result = Some(min);
+            }
+            "max" => {
+                let max = value.clone().into_f32();
+                max_result = Some(max as f64);
+            }
+            _ => {}
+        }
+    }
+
+    // For Clip Opset 11+ , the min and max values are inputs
+    // Get the min and max values from the input values
+    if min_result.is_none() && max_result.is_none() {
+        let min = &node.inputs[1].value;
+        let max = &node.inputs[2].value;
+
+        if min_result.is_none() && min.is_some() {
+            let min = min.clone().unwrap().into_scalar();
+            match min {
+                Data::Float16(min) => min_result = Some(f32::from(min) as f64),
+                Data::Float32(min) => min_result = Some(min as f64),
+                Data::Float64(min) => min_result = Some(min),
+                _ => panic!("Clip: only float min is supported"),
+            };
+        }
+
+        if max_result.is_none() && max.is_some() {
+            let max = max.clone().unwrap().into_scalar();
+            match max {
+                Data::Float16(max) => max_result = Some(f32::from(max) as f64),
+                Data::Float32(max) => max_result = Some(max as f64),
+                Data::Float64(max) => max_result = Some(max),
+                _ => panic!("Clip: only float max is supported"),
+            };
```

Comment:
It's easier to read if the `max_result` is assign before the match:

```rust
max_result = match max {
                Data::Float16(max) => Some(f32::from(max) as f64),
                Data::Float32(max) =>  Some(max as f64),
                Data::Float64(max) =>   Some(max),
                _ => panic!("Clip: only float max is supported"),
            };
```

Same for min.

Reply:
Thanks. It's fixed. It probably evolved into this =)